packages/calypso-e2e/src/lib/blocks/image-block.ts

```diff
@@ -79,17 +78,17 @@ export class ImageBlock {
 	 *
 	 * @param {string} path Path to the image file.
 	 */
-	async uploadFromModal( path: string ): Promise< ElementHandle > {
-		const frame = ( await this.block.ownerFrame() ) as Frame;
-		const page = frame.page();
+	async uploadThroughMediaLibrary( path: string ): Promise< ElementHandle > {
+		await this.selectImageSource( 'Media Library' );
 
-		// The resulting react modal embeds the MediaPage with additional
-		// buttons to confirm/cancel action.
-		const reactModalComponent = new ReactModalComponent( page, new MediaPage( page ) );
-		await reactModalComponent.pageObject.upload( path );
-		await reactModalComponent.clickButton( 'Insert' );
+		const page = ( await this.block.ownerFrame() )?.page() as Page;
+
+		await page.locator( 'input[type="file"][multiple]' ).setInputFiles( path );
+		const confirmButtonSelector = envVariables.TEST_ON_ATOMIC
+			? 'button:text-is("Select")'
+			: 'button :text-is("Insert")'; // The whitespace is intentional as the text is nested in a span element.
+		await page.locator( confirmButtonSelector ).click();
```

Comment:
**comment/blocking**: I'm wondering if racing two selectors would do here, like [this](https://playwright.dev/docs/selectors#css-selector-list):

```
page.locator( 'button:text-is("Select"), button :text-is("Insert")')
```

Or, would making the core of the selector the following and changing only the text to match on:

```
button :text-is("${ text }")
```

Reply:
It would be neater code-wise, but I made that distinction on purpose so we know that it comes from an Atomic vs Simple difference. I guess we could also just make a comment instead of applying the env var, but I think it's still better that the code is self-explanatory.